### ISSUE TITLE
Prove certain hints with ghc-proofs

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#309]
https://github.com/nomeata/ghc-proofs/ should be able to prove certain hints are correct by construction. I should enable that where I can.

**Comment by nomeata** [comment]
Good idea, could have been mine. I'll share my successes so far soonish (traveling now)
